validator/server/src/main/java/org/ethereum/beacon/validator/api/ValidatorRest.java

```diff
@@ -0,0 +1,389 @@
+import org.ethereum.beacon.validator.api.model.ValidatorDutiesResponse;
+import org.ethereum.beacon.wire.WireApiSub;
+import org.ethereum.beacon.wire.sync.SyncManager;
+import org.javatuples.Pair;
+import org.javatuples.Triplet;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.server.HttpServer;
+import reactor.netty.http.server.HttpServerRequest;
+import reactor.netty.http.server.HttpServerResponse;
+import tech.pegasys.artemis.util.bytes.Bytes48;
+import tech.pegasys.artemis.util.bytes.Bytes96;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * REST service for beacon chain validator according to <a
+ * href="https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/validator/beacon_node_oapi.yaml">https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/validator/beacon_node_oapi.yaml</a>
+ */
+public class ValidatorRest extends ReactorNettyServer {
+
+  private final BeaconChainSpec spec;
+  private final WireApiSub wireApiSub;
+  private final MutableBeaconChain beaconChain;
+  private final ValidatorDutiesService validatorDutiesService;
+  private final UInt64 chainId;
+
+  private ObjectMapper mapper = new ObjectMapper();
+  private String version = null;
+  private Time time = null;
+  private SyncingResponse syncingResponse = null;
+  private boolean shortSync = false;
+  private ObservableBeaconState observableBeaconState = null;
+
+  public ValidatorRest(
+      String serverHost,
+      Integer serverPort,
+      BeaconChainSpec spec,
+      ObservableStateProcessor stateProcessor,
+      SyncManager syncManager,
+      UInt64 chainId,
+      ValidatorDutiesService validatorDutiesService,
+      WireApiSub wireApiSub,
+      MutableBeaconChain beaconChain) {
+    super(serverHost, serverPort);
+    this.spec = spec;
+    this.wireApiSub = wireApiSub;
+    this.beaconChain = beaconChain;
+    this.chainId = chainId;
+    this.validatorDutiesService = validatorDutiesService;
+
+    Flux.from(stateProcessor.getObservableStateStream())
+        .subscribe(
+            observableBeaconState -> {
+              this.observableBeaconState = observableBeaconState;
+              if (time == null) {
+                time = observableBeaconState.getLatestSlotState().getGenesisTime();
+              }
+            });
+
+    Flux.from(syncManager.getSyncStatusStream())
+        .subscribe(
+            syncStatus -> {
+              shortSync = SyncManager.SyncMode.Short.equals(syncStatus.getSyncMode());
+              syncingResponse =
+                  new SyncingResponse(
```

Comment:
We might want to have a `create` method that creates `SyncingResponse` from `SyncStatus` object. It would help to keep code clean.